ctapipe/calib/camera/r1.py

```diff
@@ -191,6 +192,144 @@ def calibrate(self, event):
                 event.r1.tel[telid].waveform = calibrated
 
 
+class LSTR1Calibrator(CameraR1Calibrator):
+
+    pedestal_path = Unicode(
+        '',
+        allow_none=True,
+        help='Path to the LST pedestal file'
+    ).tag(config=True)
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        """
+        The R1 calibrator for LST data.
+        Fills the r1 container.
+        Parameters
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool : ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        kwargs
+        """
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        self.telid = 0
+        self.pedestal_value_array = None
+        self.n_pixels = 7
+        self.size4drs = 4 * 1024
+        self.roisize = 40
+        self.offset = 300
+        self.high_gain = 0
+        self.low_gain = 1
+
+        self._load_calib()
+
+    def calibrate(self, event):
+        """
+        Perform calibration on event using pedestal file.
+        If pedestal file is wrong raise ValueError Exception.
+        Parameters
+        ----------
+        event : `ctapipe` event-container
+        """
+        samples = np.zeros((event.r0.tel[self.telid].waveform.shape), dtype=np.uint16)
+        event.r1.tel[self.telid].waveform = samples
+
+        event_number_of_modules = event.lst.tel[0].svc.num_modules
+        if event_number_of_modules == self.number_of_modules_from_file:
+            for nr_module in range(0, event_number_of_modules):
+                first_cap = self._get_first_capacitor(event, nr_module)
+                for gain in range(0, 2):
+                    for pixel in range(0, self.n_pixels):
+                        for roi in range(0, self.roisize):
+                            position = int((roi + first_cap[gain, pixel]) % self.size4drs)
+                            val = (
+                                event.r0.tel[0].waveform[gain, pixel + nr_module * 7, roi]
+                                - int(self.pedestal_value_array[nr_module, gain,
```

Comment:
This code is very non-pythonic (it's written more like C), and will thus be very slow - it should be done using numpy vector operations, and that would probably simplify it to only a few lines.

Reply:
Thanks for feedback. Now, I known, I am working on it, soon I will send changes. What do you think about using `numba` to speed-up code?